src/Symfony/Component/Utf8/CodePoints.php

```diff
@@ -0,0 +1,234 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Utf8;
+
+use Symfony\Component\Utf8\Exception\InvalidArgumentException;
+
+final class CodePoints implements GenericStringInterface, \Countable
+{
+    use Utf8Trait;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function count(): int
+    {
+        return mb_strlen($this->string, 'UTF-8');
```

Comment:
Yes, my bad, I just skimmed your comment. The whole thing is really not a big deal, because the string is not really magic (I hope everyone knows what `UTF-8` is), and limited on a single component.